src/containers/oracle.js

```diff
@@ -396,26 +384,57 @@ class OraclePage extends React.Component {
     }
   }
 
-  checkAllowance() {
-    try {
-      const { selectedWalletAddress } = this.props;
+  /**
+ * Send get allowance request and keep repeating itself until this.state.isApproving is false
+ * @return {[type]}
+ */
+  startCheckAllowance() {
+    console.log('startCheckAllowance', new Date());
+    const { selectedWalletAddress, onAllowance } = this.props;
+    const { oracle } = this.state;
+
+    const self = this;
 
-      this.props.onAllowance(selectedWalletAddress, this.state.oracle.topicAddress, selectedWalletAddress);
-    } catch (err) {
-      console.log(err.message);
+    // A function to repeat itself until this.state.isApproving is false
+    function startPollAllowance() {
+      if (self.state.isApproving) {
+        onAllowance(selectedWalletAddress, oracle.topicAddress, selectedWalletAddress);
+        setTimeout(startPollAllowance, ALLOWANCE_TIMER_INTERVAL);
```

Comment:
If you don't set this to a var, you can't cancel the timer when you leave the page. Or did you intentionally have it still running in the background when you leave the page? 

If so, this could be tricky say if you try to set result on 3 different Oracles. Then there would be 3 timers running in the background.

Reply:
I think you mistook setTimeout() for setInterval(). The former only set timer once and needs to call it self in execution block to repeat. When SetResult() or Approve(), isApproving is set to false, so timer is not set again in the last round.

Reply:
oops forgot to remove this comment. sorry, ya I tested it out and there was only 1 timer.